File: MS.NET/Sarvam Backend/Models/MasterAttendance.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Sarvam.Models
{
    [Table("master_attendance")]
    public class MasterAttendance
    {
        [Key]
        [Column("attendance_id")]
        public int AttendanceId { get; set; }

        [ForeignKey("MasterProfile")]
        [Column("PRN")]
        public long PRN { get; set; }

        [Required]
        [Column("date")]
        public DateOnly Date { get; set; }

        [Required]
        [Column("status")]
        public bool Status { get; set; }

        // Adding created_at and updated_at columns
        [Required]
        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;  // Default value set to current time

        [Required]
        [Column("updated_at")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;  // Default value set to current time

        // Mark the navigation property as optional
        public MasterProfile? MasterProfile { get; set; }
    }
}

File: Sarvam C-DAC Project/Sarvam C-DAC Spring Boot/Sarvam/src/main/java/com/cdac/sarvam/model/MasterAttendance.java
package com.cdac.sarvam.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import jakarta.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "master_attendance")
public class MasterAttendance {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "attendance_id")
    private Long attendanceId;

    @ManyToOne
    @JoinColumn(name = "PRN", referencedColumnName = "PRN", nullable = false)
    private MasterProfile masterProfile; // Foreign key reference to MasterProfile

    @Column(name = "date", nullable = false)
    private LocalDate date;

    @Column(name = "status", nullable = false)
    private boolean status;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt = LocalDateTime.now();
    
    public void setProfile(MasterProfile profile) {  // This is the missing setter method
        this.masterProfile = profile;
    }
}
